drydock_provisioner/drydock_client/client.py

```diff
@@ -14,83 +14,213 @@
 import json
 import requests
 
-from .session import DrydockSession
+from drydock_provisioner import error as errors
 
 class DrydockClient(object):
     """"
     A client for the Drydock API
 
-    :param string host: Hostname or IP address of Drydock API
-    :param string port: Port number of Drydock API
-    :param string version: API version to access
-    :param string token: Authentication token to use
-    :param string marker: (optional) External marker to include with requests
+    :param DrydockSession session: A instance of DrydockSession to be used by this client
+    :param string version: Drydock API version to use, default is '1.0'
     """
 
-    def __init__(self, host=None, port=9000, version='1.0', token=None, marker=None):
+    def __init__(self, session, version='1.0'):
+        self.session = session
         self.version = version
-        self.session = DrydockSession(token=token, ext_marker=marker)
-        self.base_url = "http://%s:%d/api/%s/" % (host, port, version)
+        self.api_url = "%s/%s/" % (self.session.base_url, version)
 
-    def send_get(self, api_url, query=None):
+    def __send_get(self, endpoint, query=None):
```

Comment:
Also, this is a matter of personal opinion. But the double-underscore prefix is a way of say "I really don't want this imported by anyone else ever" and Python interpreter does some mangling to make this harder to use elsewhere. But typically the convention for private methods is to use a single-underscore prefix, e.g.:

```python

def __super_private_method(self):
    ...

def _typical_private_method(self):
    ...
```